tools/flaky-test-retryer/handler.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"log"
+
+	// TODO: remove this import once "k8s.io/test-infra" import problems are fixed
+	// https://github.com/test-infra/test-infra/issues/912
+	"github.com/knative/test-infra/tools/monitoring/prowapi"
+)
+
+// TODO(TrevorFarrelly): supported repos depend on flaky-test-reporter's
+// configuration. Use the reporter's config.yaml to build this list.
+var repos = []string{"serving"}
+
+// validStatus checks that the message signals a failure
+func validStatus(msg *prowapi.ReportMessage) bool {
+	return msg.Status == prowapi.FailureState
+}
+
+// validType checks that the message came from a presubmit job
+func validType(msg *prowapi.ReportMessage) bool {
+	return msg.JobType == prowapi.PresubmitJob
+}
+
+// validRepo checks that the message came from a repo the reporter runs on
+func validRepo(msg *prowapi.ReportMessage) bool {
+	if len(msg.Refs) > 0 {
+		for _, repo := range repos {
+			if msg.Refs[0].Repo == repo {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// handleDriver spawns a goroutine for each valid message received
+func handleDriver(msg *prowapi.ReportMessage) {
+	if validStatus(msg) && validType(msg) && validRepo(msg) {
```

Comment:
Alternatively, you can merge all these 3 into a single function like `expectedMsg`, instead of addressing my comments above